### ISSUE TITLE
No kotlin lines and code smells detected after updating to 0.2.4 on Sonarqube 6.5

### ISSUE THREAD

**Issue by JDurstberger** [detekt/sonar-kotlin#33]
Hello, 

I hope I did not miss anything from the docs, but after updating from 0.2.3 to 0.2.4 we 
do not see any LOCs or code smells reported for the kotlin files anymore. 

I looked at the release notes and the git history and found, that are now using 
`detekt` for these. 
Do we need to make any additional changes to our projects to get these reports back?
Do we need to run detekt locally before triggering the sonarqube task?

Thanks in advance

**Comment by arturbosch** [comment]
That's strange! `sonar-kotlin` always uses `detekt` for the analysis. It's just the version update.

I just started with a new db for sonar and run an analysis on detekt itself. I get code smells and loc data. 
Which version of sonar do you use?

**Comment by Altoyyr** [comment]
We are running on Sonarqube 6.5.

Could this be an update issue, if you are saying it works fine on a fresh project?

**Comment by arturbosch** [comment]
Just run again on detekt with the old database. Got three new code smells. I'm running on 6.4, will check out 6.5.

**Comment by Altoyyr** [comment]
Thank you :)

**Comment by cocorossello** [comment]
I am on sonarqube 6.5 and I have kotlin lines/smells in 0.2.3 and 0.2.4 (after I upgraded), if that helps.

I installed kotlin plugin after I updgraded to 6.5 (I have no idea if that relates in any way)

**Comment by arturbosch** [comment]
When downloading a new sonar 6.5 than adding the kotlin plugin and than analyzing everything is there. Yesterday I installed 6.5, run analysis on a project without the kotlin plugin, than added the plugin and run again -> no data.
@cocorossello you had previous data for your project before updating?

**Comment by Altoyyr** [comment]
We had sonar 6.5 with 0.2.3 and a bit over a month of data before 
we updated to 0.2.4. 

So it seems like a migration issue?

**Comment by arturbosch** [comment]
I guess so. 
I don't really know the interns of sonar good enough. But as detekt itself is not 1.0.GA and some rules may change in description and names, this can effect the plugin. I haven't encountered a migration problem so far (from 0.2.0 to 0.2.4) with my instance and my projects.

Have you tried a new sonar instance with the new version? Or a new instance with your db and the new plugin version

**Comment by cocorossello** [comment]
FYI: I just upgraded to 6.6 and 6.7 later (from 6.5) and I didn't have any issue.

**Comment by Altoyyr** [comment]
@cocorossello Did you just update sonar or also the plugin?

**Comment by sirago** [comment]
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.3.0.603:sonar (default-cli) on project Patent_Import: io.gitlab.arturbosch.detekt.sonar.jacoco.KotlinJaCoCoSensor has unsatisfied dependency 'interface org.sonar.plugins.java.api.JavaResourceLocator' for constructor 'public io.gitlab.arturbosch.detekt.sonar.jacoco.KotlinJaCoCoSensor(org.sonar.api.batch.fs.FileSystem,org.sonar.plugins.java.api.JavaResourceLocator,org.sonar.java.JavaClasspath)' from org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@50eb4a2c:133<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@7dbe2ebf:216<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@1aabf50d:7<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@11d86b9d:30<|

I'm getting that with Sonar 6.7

**Comment by legendjaks** [comment]
@sirago with sonar-kotlin tag 0.2.4 (not master) and sonarqube 6.5, no issues.

**Comment by cocorossello** [comment]
I only updated sonar, 2 times. Also I'm not get getting @sirago 's error on 6.7

**Comment by mike-stetsenko** [comment]
The same problem. Sonarqube 6.7, plugin 0.2.4, kotlin LOC is zero.

**Comment by arturbosch** [comment]
Just tried using sonar 6.7 LTS, sonar-kotlin build from master and get code smells and size metrics. Someone can confirm the master build?

**Comment by Altoyyr** [comment]
I think this is a configuration issue, we are getting LOCS, codesmells and coverage after changing our config to

```
sonarqube {
    properties {
        property "sonar.host.url", --OMITTED--
        property "sonar.login", --OMITTED--
        property 'sonar.projectName',--OMITTED--
        property 'sonar.projectKey',--OMITTED--
        property 'sonar.projectVersion',--OMITTED--
        property 'sonar.sourceEncoding', 'UTF-8'

        property "sonar.java.binaries", "build/intermediates/classes, build/tmp/kotlin-classes"
        property "sonar.exclusions", "src/main/res/**/*"

        property 'sonar.junit.reportPaths', "build/test-results/testDebugUnitTest, build/test-results/testReleaseUnitTest, build/test-results/test"
        property "sonar.java.coveragePlugin", "jacoco"
        property "sonar.jacoco.reportPaths", "build/jacoco/testDebugUnitTest.exec, build/jacoco/testReleaseUnitTest.exec, build/jacoco/test.exec"
    }
}
```
Hope this helps other people.

The only issue we are now seeing is, that it does NOT report kotlin files when running sonarqube on travis, it works fine though when run on a local machine. 

I guess this could be a separate issue, but I think this is also a config issue on our side.

**Comment by Ishaks** [comment]
Is there a URL available for plugin version compatible with 6.7(https://someurl) we can use in docker container to build an image?

**Comment by arturbosch** [comment]
https://github.com/arturbosch/sonar-kotlin/releases/tag/0.4.1
I do not have any problems with 6.7 and this version.
